pkg/issuer/acme/dns/dns.go

```diff
@@ -141,22 +142,35 @@ func (s *Solver) solverFor(crt *v1alpha1.Certificate, domain string) (solver, er
 			return nil, fmt.Errorf("error instantiating cloudflare challenge solver: %s", err.Error())
 		}
 	case providerConfig.Route53 != nil:
-		secretAccessKeySecret, err := s.secretLister.Secrets(s.resourceNamespace).Get(providerConfig.Route53.SecretAccessKey.Name)
-		if err != nil {
-			return nil, fmt.Errorf("error getting route53 secret access key: %s", err.Error())
-		}
+		secretAccessKey := ""
+		if providerConfig.Route53.SecretAccessKey.Name != "" {
+			secretAccessKeySecret, err := s.secretLister.Secrets(s.resourceNamespace).Get(providerConfig.Route53.SecretAccessKey.Name)
+			if err != nil {
+				return nil, fmt.Errorf("error getting route53 secret access key: %s", err.Error())
+			}
 
-		secretAccessKeyBytes, ok := secretAccessKeySecret.Data[providerConfig.Route53.SecretAccessKey.Key]
-		if !ok {
-			return nil, fmt.Errorf("error getting route53 secret access key: key '%s' not found in secret", providerConfig.Route53.SecretAccessKey.Key)
+			secretAccessKeyBytes, ok := secretAccessKeySecret.Data[providerConfig.Route53.SecretAccessKey.Key]
+			if !ok {
+				return nil, fmt.Errorf("error getting route53 secret access key: key '%s' not found in secret", providerConfig.Route53.SecretAccessKey.Key)
+			}
+			secretAccessKey = string(secretAccessKeyBytes)
 		}
 
-		impl, err = route53.NewDNSProviderAccessKey(
-			providerConfig.Route53.AccessKeyID,
-			string(secretAccessKeyBytes),
-			providerConfig.Route53.HostedZoneID,
-			providerConfig.Route53.Region,
-		)
+		if s.ambientCredentials {
+			impl, err = route53.NewDNSProviderAccessKey(
+				providerConfig.Route53.AccessKeyID,
+				secretAccessKey,
+				providerConfig.Route53.HostedZoneID,
+				providerConfig.Route53.Region,
+			)
+		} else {
+			impl, err = route53.NewDNSProviderAmbient(
+				providerConfig.Route53.AccessKeyID,
+				secretAccessKey,
+				providerConfig.Route53.HostedZoneID,
+				providerConfig.Route53.Region,
+			)
```

Comment:
So I'm thinking in order to test this, we could probably just add a unit test that ensures the correct function is called (i.e. ambient or not). We will struggle otherwise to run these tests, as we currently don't mock the entire provider environment (and thus, we have no way to properly e2e test DNS providers at the moment either).